tests/scene/test_path_2d.h

```diff
@@ -0,0 +1,109 @@
+/* Copyright (c) 2007-2022 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2022 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef TEST_PATH_2D_H
+#define TEST_PATH_2D_H
+
+#include "scene/2d/path_2d.h"
+
+#include "tests/test_macros.h"
+
+namespace TestPath2D {
+
+TEST_CASE("[SceneTree][Path2D] Initialization") {
+	SUBCASE("Path should be empty right after initialization") {
+		Path2D *test_path = memnew(Path2D);
+		CHECK(test_path->get_curve() == nullptr);
+		memdelete(test_path);
+	}
+}
+
+TEST_CASE("[SceneTree][Path2D] Curve setter and getter") {
+	Path2D *test_path = memnew(Path2D);
+	const Ref<Curve2D> &curve = memnew(Curve2D);
+
+	SUBCASE("Curve passed to the class should remain the same") {
+		test_path->set_curve(curve);
+		CHECK(test_path->get_curve() == curve);
+	}
+	SUBCASE("Curve passed many times to the class should remain the same") {
+		test_path->set_curve(curve);
+		test_path->set_curve(curve);
+		test_path->set_curve(curve);
+		CHECK(test_path->get_curve() == curve);
+	}
+	SUBCASE("Curve rewrite testing") {
+		const Ref<Curve2D> &curve1 = memnew(Curve2D);
+		const Ref<Curve2D> &curve2 = memnew(Curve2D);
+
+		test_path->set_curve(curve1);
+		test_path->set_curve(curve2);
+		CHECK_MESSAGE(test_path->get_curve() != curve1,
+				"After rewrite, second curve should be in class");
+		CHECK_MESSAGE(test_path->get_curve() == curve2,
+				"After rewrite, second curve should be in class");
```

Comment:
This wordings seems a bit off to me, but that's also used in the Path3D tests, so I think we can change that later.